README.md

```diff
@@ -25,3 +25,216 @@
+    section will be considered as relative to this directory.
+  - **variable_default**: Allows one to specify defaults for all variables in
+    the tested_variables section so that those defaults need not be repeated
+    multiple times.
+  - **variable**: Each entry in this section is the name of a tested variable
+    (such as "ksads_1_187_t"). Associated with it is information on how to treat
+    the variable.
+    - **filename**: ABCD csv file that contains the data for the tested variable
+    - **convert**: A dictionary that explains how to convert the tested
+      variable's data. Each (key, value) pair of the dictionary specifies what
+      the datum (the key) is to be converted to (the value).
+    - **handle_missing**: Can be set to any of
+      - **invalidate**: throw away target image if it has this field missing
+      - **together**: all target images marked as "missing" are put in one
+        category that is dedicated to all missing data
+      - **by_value**: for each `is_missing` value, all target images with that
+        value are put in a category
+      - **separately**: each row with a missing value is its own category; e.g.,
+        a patient with no siblings in the study
+    - **type**: whether the variable is "ordered" (such as a number representing
+      an intensity) or "unordered" (as in a number representing a category)
+    - **is_missing**: a list of values that should be considered as missing such
+      as `.nan` or `""`
+    - **description**: an optional, user-supplied string
+    - **internal_name**: optionally, the name by which this tested variable will
+      be known within `filename`.
+- **target_variables**: This section specifies the target images that are to be
+  analyzed. The specific fields are:
+  - **source_directory**: All relative paths supplied in the target_variables
+    section will be considered as relative to this directory.
+  - **table_of_filenames_and_metadata**: the file location for the CSV file that
+    lists target images to be analyzed. The columns of the CSV file are
+    `filename`, `src_subject_id`, `event_name`, `modality`, and `description`,
+    as described next.
+  - **individual_filenames_and_metadata**: a way to specify target images in the
+    YAML file (rather than in the CSV file).
+    - **filename**: the location of the .nii.gz file containing the voxel
+      intensities
+    - **src_subject_id**: the ABCD subject ID associated with this file (with
+      prefix `NDAR_`).
+    - **event_name**: the ABCD event_name associated with this file (such as
+      `2_year_follow_up_y_arm_1`).
+    - **modality**: Such as `fa` or `md`
+    - **description**: an optional, user-supplied string
+  - **mask**: describes mask (e.g., white matter mask) that specifies which
+    voxels are meaningful
+    - **filename**: the location of the .nii.gz file containing the mask
+    - **threshold**: the threshold intensity for determinging which voxels to
+      include
```

Comment:
I think the thresholding shouldn't be the job of abcdstats -- instead the user should provide a boolean mask. They may have obtained that mask by thresholding or by some other method, but it's up to the user to do that. So I think it should just be a binary mask image here.

That being said, if you threshold a binary mask image that takes values 0 and 1 at a threshold of, say, 0.5 then you get the same image back, so maybe this is fine if the default value of `threshold` is something between 0 and 1. The user can omit `threshold` in that case and the abcdstats internal thresholding would be a no-op

Reply:
`nilearn.maskers.NiftiMasker` does some sort of thresholding, but there is a decent chance that it also does memory management.  So, even though we could do the masking ourselves, I am thinking to let `nilearn` do it for us.  That way we get all the optimizations that `nilearn` has and will add into the future.  And, as you suggest, we'll default the threshold to `0.5` to make it as easy as possible for the user in the case that the user doesn't want the help from `nilearn`.

Reply:
Actually ... I think that `nilearn.maskers.NiftiMasker` helps with memory management for the loading of the target images input only in the process of masking.  So, maybe not all whole brains have to be in memory at once, but I do believe that all images' retained voxels (that are not masked away) will have to be in memory at once.

Additionally, _maybe_ the masker does memory management in interpreting the (much smaller) output from `nilearn.mass_univariate.permuted_ols`; that is, it is `masker.inverse_transform(permuted_ols_response['logp_max_t'])` that is of type `nibabel.nifti1.Nifti1Image` and thus might possibly somehow take advantage of lazy loading for nifti images.